tasks/kernel_matrix_testing/ci.py

```diff
@@ -231,9 +226,10 @@ def get_all_jobs_for_pipeline(pipeline_id: int | str) -> tuple[list[KMTSetupEnvJ
     setup_jobs: list[KMTSetupEnvJob] = []
     test_jobs: list[KMTTestRunJob] = []
 
-    gitlab = get_gitlab()
-    for job in gitlab.all_jobs(pipeline_id):
-        name = job.get("name", "")
+    gitlab = get_gitlab_repo()
+    jobs = gitlab.pipelines.get(pipeline_id, lazy=True).jobs.list(per_page=100, all=True)
+    for job in jobs:
+        name = job.name
         if name.startswith("kmt_setup_env"):
             setup_jobs.append(KMTSetupEnvJob(job))
         elif name.startswith("kmt_run_"):
```

Comment:
Here you'd need to pass the `gitlab` object (Github doesn't let me make a suggestion here :( )